File: OCH_Tao/tessoku-book/A11.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [N, X] = input[0].split(" ").map(x => parseInt(x, 10));
const A = input[1].split(" ").map(x => parseInt(x, 10));
let L = 0;
let R = N - 1;
while (L <= R) {
  let M = Math.floor((L + R) / 2);
  if (X === A[M]) {
    console.log(M + 1);
    break;
  }
  if (X < A[M]) {
    R = M - 1;
  }
  if (X > A[M]) {
    L = M + 1;
  }
}

File: OCH_Tao/tessoku-book/A11.py
N, X = map(int, input().split())
A = list(map(int, input().split()))
L = 0
R = N-1
while L <= R:
  M = (L+R)//2
  if X == A[M]:
    print(M+1)
    break
  if X < A[M]:
    R = M-1
  if X > A[M]:
    L = M+1
